### PR TITLE
feat(workspace): Add skeleton workspace data source

### PR DESCRIPTION
**Pull request by cameronwaterman**
# Pull Request

## 🤨 Rationale

https://dev.azure.com/ni/DevCentral/_workitems/edit/2527387

## 👩‍💻 Implementation

- `npm run generate`
- Add to provisioning example file
- Fill in README
- Update data source base and test routes

## 🧪 Testing

Validated the datasource is available locally

## ✅ Checklist

- [X] This PR has a title that follows the [commit message format](https://github.com/ni/systemlink-grafana-plugins#commit-message-format).

### CODE REVIEW

File: src/datasources/workspace/WorkspaceDataSource.test.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+} from 'test/fixtures';
+import { WorkspaceDataSource } from './WorkspaceDataSource';
+
+let ds: WorkspaceDataSource, backendSrv: MockProxy<BackendSrv>;
+
+beforeEach(() => {
+  [ds, backendSrv] = setupDataSource(WorkspaceDataSource);
+});
+
+fdescribe('testDatasource', () => {
```
Comment (mure) [inline src/datasources/workspace/WorkspaceDataSource.test.ts:18]:
```suggestion
describe('testDatasource', () => {
```

**Review by mure — CHANGES_REQUESTED** [review]

File: src/datasources/workspace/WorkspaceDataSource.test.ts (hunk 1 above)
Comment (mure) [reply src/datasources/workspace/WorkspaceDataSource.test.ts:18]: I should see if I can update the linting rules to catch these

Comment (cameronwaterman) [reply src/datasources/workspace/WorkspaceDataSource.test.ts]: good catch

**Review by mure — APPROVED** [review]

**Pull request merged**